HW0/P4/P4.py

```diff
@@ -40,37 +41,96 @@
     # Read the observation array and plot it (Part 2)
     #####################
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    s_measured = np.loadtxt('P4_measurements.txt', delimiter=',')
+
+    # store for part 4 before adjusting
+    m = s_measured
+
+    # adjust for dimensional stretching
+    s_measured = np.multiply(s_measured, np.array([1. / rx, 1. / ry, 1. / rz]))
+
+    x_coords, y_coords, z_coords = s_measured.T
+    ax.plot(x_coords, y_coords, z_coords, '.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
 
-    # A = ?
-    # a = ?
-    # s = ?
+    A = np.array([
+        [1, 0, 0, dt, 0, 0],
+        [0, 1, 0, 0, dt, 0],
+        [0, 0, 1, 0, 0, dt],
+        [0, 0, 0, 1 - c * dt, 0, 0],
+        [0, 0, 0, 0, 1 - c * dt, 0],
+        [0, 0, 0, 0, 0, 1 - c * dt]
+    ])
+
```

Comment:
you can use the np.diag() function to populate a diagonal matrix more efficiently.